lib/apiservers/engine/backends/container.go

```diff
@@ -383,6 +383,53 @@ func (c *Container) ContainerExecStart(ctx context.Context, eid string, stdin io
 	actor := CreateContainerEventActorWithAttributes(vc, map[string]string{})
 	EventService().Log(event, eventtypes.ContainerEventType, actor)
 
+	ca := &backend.ContainerAttachConfig{
+		UseStdin:  stdin != nil,
+		UseStdout: stdout != nil,
+		UseStderr: stderr != nil,
+	}
+
+	err = c.containerProxy.AttachStreams(context.Background(), vc, eid, stdin, stdout, stderr, ca)
+	if err != nil {
+		if _, ok := err.(DetachError); ok {
+			log.Infof("Detach detected, tearing down connection")
```

Comment:
This chunk of logic wants to be moved out into a separate method, current method is heavily overloaded already.